File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Transport/AirTransport.java
package com.logistics.Components.Transport;

public class AirTransport extends Transport {
    private String model;

    // Constructors
    public AirTransport() {
        super();
        model = "";
    }

    public AirTransport(int code, int speed, float costPerMeter, String route, String model) {
        super(code, speed, costPerMeter, route);
        this.model = model;
    }

    // Copy Constructor
    public AirTransport(AirTransport other) {
        super(other);
        this.model = other.model;
    }

    // Clone method
    @Override
    public AirTransport clone() {
        return new AirTransport(this);
    }

    // Getters
    public String getModel() { return model; }

    // Setters
    public void setModel(String model) { this.model = model; }

    @Override
    public String to_string() {
        return super.to_string() +
                "\nModel: " + model;
    }
}

File: Task 1/C++/Components/Transport/AirTransport.cpp
#include "AirTransport.h"

AirTransport::AirTransport() : Transport(){
    model = "";
}

AirTransport::AirTransport(int code, float speed, float costPerMeter, const std::string route, const std::string model) 
            : Transport(code, speed, costPerMeter, route){
    this->model = model;
}

AirTransport::AirTransport(const AirTransport& other) : Transport(other) {
    this->model = other.model;
}

std::string AirTransport::to_string(){
    return Transport::to_string()+
            "\nModel: "+model;
}